src/NuGet.Clients/NuGet.PackageManagement.UI/Xamls/PackageManagerControl.xaml.cs

```diff
@@ -39,127 +40,92 @@ namespace NuGet.PackageManagement.UI
-
-        // Signifies where an action is being executed. Should be updated in a coordinated fashion with IsEnabled
-        private bool _isExecutingAction;
+        private bool _isExecutingAction; // Signifies where an action is being executed. Should be updated in a coordinated fashion with IsEnabled
         private RestartRequestBar _restartBar;
-
-        public PackageRestoreBar RestoreBar { get; private set; }
-
         private PRMigratorBar _migratorBar;
-
-        private readonly IVsWindowSearchHost _windowSearchHost;
-        private readonly IVsWindowSearchHostFactory _windowSearchHostFactory;
-
-        internal readonly DetailControlModel _detailModel;
-
         private bool _missingPackageStatus;
-        private readonly INuGetUILogger _uiLogger;
         private bool _loadedAndInitialized = false;
         private bool _recommendPackages = false;
-
         private (string modelVersion, string vsixVersion)? _recommenderVersion;
 
-        public PackageManagerModel Model { get; }
-
-        public ISettings Settings { get; }
-
-        internal ItemFilter ActiveFilter { get => _topPanel.Filter; set => _topPanel.SelectFilter(value); }
-
-        internal InfiniteScrollList PackageList { get => _packageList; }
-
-        internal PackageSourceMoniker SelectedSource
+        private PackageManagerControl()
         {
-            get
-            {
-                return _topPanel.SourceRepoList.SelectedItem as PackageSourceMoniker;
-            }
-            set
-            {
-                _topPanel.SourceRepoList.SelectedItem = value;
-            }
+            InitializeComponent();
         }
 
-        internal IEnumerable<PackageSourceMoniker> PackageSources => _topPanel.SourceRepoList.Items.OfType<PackageSourceMoniker>();
-
-        internal IEnumerable<SourceRepository> ActiveSources => SelectedSource?.SourceRepositories ?? Enumerable.Empty<SourceRepository>();
-
-        internal event EventHandler _actionCompleted;
-
-        public bool IncludePrerelease => _topPanel.CheckboxPrerelease.IsChecked == true;
-
-        private readonly Guid _sessionGuid = Guid.NewGuid();
-        private readonly Stopwatch _sinceLastRefresh;
-        private bool _installedTabDataIsLoaded;
-        private bool _updatesTabDataIsLoaded;
+        public static Task<PackageManagerControl> CreateAsync(PackageManagerModel model, INuGetUILogger uiLogger)
+        {
+            Assumes.NotNull(model);
 
-        private bool _forceRecommender;
+            var ret = new PackageManagerControl();
+            return ret.InitializeAsync(model, uiLogger);
+        }
 
-        public PackageManagerControl(
-            PackageManagerModel model,
-            ISettings nugetSettings,
-            IVsWindowSearchHostFactory searchFactory,
-            IVsShell4 vsShell,
-            INuGetUILogger uiLogger = null)
+        private async Task<PackageManagerControl> InitializeAsync(PackageManagerModel model, INuGetUILogger uiLogger)
```

Comment:
The return type can simply be `Task` (or maybe even `ValueTask`).

Reply:
I did it this way so that I could just do "return packageManagerControl.InitializeAsync" instead of 
"await packageManagerControl.InitializeAsync
 return packageManagerControl"
I will change it to what you suggested.